src/Components/Components/src/Routing/OnNavigateArgs.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Routing
+{
+    public class OnNavigateArgs
```

Comment:
On naming, it would be more common to see this as something like `NavigateEventArgs`. However I'm aware that this isn't strictly a C# `event` so would agree there's more flexibility than usual with how this is named.

Javier previously mentioned `NavigationContext` as a possible name, which also sounds good. What do you think?

Reply:
Dropping the "on" preposition is fine with me. I have no strong preference against or for the names. I gravitated towards `NavigateArgs` to match the `LocationChangedEventArgs` used in the NavigationManager.

The OnNavigate isn't really an event but it also sort of is.